lightning/src/util/events.rs

```diff
@@ -201,22 +229,31 @@ impl MaybeReadable for Event {
 				let f = || {
 					let mut payment_hash = PaymentHash([0; 32]);
 					let mut payment_preimage = None;
-					let mut payment_secret = PaymentSecret([0; 32]);
+					let mut payment_secret = None;
 					let mut amt = 0;
-					let mut user_payment_id = 0;
+					let mut user_payment_id = None;
 					read_tlv_fields!(reader, {
 						(0, payment_hash, required),
-						(2, payment_secret, required),
+						(2, payment_secret, option),
```

Comment:
I think there's ways to make both directions work depending on if you're willing to add a special case in the code for backwards-compatibility? @TheBlueMatt would have to confirm

Reply:
I think going from option to required is never allowed. If we move from option to required, and an old version "wrote" `None`, we'll fail to deserialize data written by an old node, which is not allowed unless we are deliberately breaking reading old versions (presumably after several sequential versions that always wrote `Some` so that you have to be reading very old data to fail).